binding.c

```diff
@@ -1369,6 +1374,113 @@ bare_ffmpeg_scaler_scale(js_env_t *env, js_callback_info_t *info) {
   return result;
 }
 
+static js_value_t *
+bare_ffmpeg_dictionary_init(js_env_t *env, js_callback_info_t *info) {
+  int err;
+  js_value_t *handle;
+
+  bare_ffmpeg_dictionary_t *dict;
+  err = js_create_arraybuffer(env, sizeof(bare_ffmpeg_dictionary_t), (void **) dict, &handle);
+
+  dict->handle = NULL;
+
+  return handle;
+}
+
+static js_value_t *
+bare_ffmpeg_dictionary_set(js_env_t *env, js_callback_info_t *info) {
+  int err;
+
+  size_t argc = 3;
+  js_value_t *argv[3];
+
+  err = js_get_callback_info(env, info, &argc, argv, NULL, NULL);
+  assert(err == 0);
+  assert(argc == 3);
+
+  bare_ffmpeg_dictionary_t *dict;
+  err = js_get_arraybuffer_info(env, argv[0], (void **) dict, NULL);
+  assert(err == 0);
+
+  size_t len;
+  err = js_get_value_string_utf8(env, argv[1], NULL, 0, &len);
+  assert(err == 0);
+
+  len += +1 /* NULL */;
+
+  utf8_t *key = malloc(len);
+  err = js_get_value_string_utf8(env, argv[1], key, len, NULL);
+  assert(err == 0);
+
+  err = js_get_value_string_utf8(env, argv[2], NULL, 0, &len);
+  assert(err == 0);
+
+  len += +1 /* NULL */;
+
+  utf8_t *value = malloc(len);
+  err = js_get_value_string_utf8(env, argv[2], value, len, NULL);
+  assert(err == 0);
+
+  err = av_dict_set(&dict->handle, (const char *) key, (const char *) value, 0);
+  assert(err);
+
+  return NULL;
+}
+
+static js_value_t *
+bare_ffmpeg_dictionary_get(js_env_t *env, js_callback_info_t *info) {
+  int err;
+
+  size_t argc = 2;
+  js_value_t *argv[2];
+
+  err = js_get_callback_info(env, info, &argc, argv, NULL, NULL);
+  assert(err == 0);
+  assert(argc == 2);
+
+  bare_ffmpeg_dictionary_t *dict;
+  err = js_get_arraybuffer_info(env, argv[0], (void **) dict, NULL);
+  assert(err == 0);
+
+  size_t len;
+  err = js_get_value_string_utf8(env, argv[1], NULL, 0, &len);
+  assert(err == 0);
+
+  len += +1 /* NULL */;
+
+  utf8_t *key = malloc(len);
+  err = js_get_value_string_utf8(env, argv[1], key, len, NULL);
+  assert(err == 0);
+
+  AVDictionaryEntry *entry = av_dict_get(dict->handle, (const char *) key, NULL, 0);
+  assert(entry != NULL);
+
+  js_value_t *result;
+  err = js_create_string_utf8(env, (const utf8_t *) entry->value, sizeof(entry->value), &result);
```

Comment:
```suggestion
  err = js_create_string_utf8(env, (const utf8_t *) entry->value, -1, &result);
```

`sizeof(entry->value)` will give you the size of `char *`, i.e. the size of a pointer, and not the size of the string value.

Reply:
Yeah I confused with `strlen()` I guessed haha. Shot!